### DISCUSSION TITLE
User-defined data type translation to OWL

### DISCUSSION THREAD

**Discussion by AbhaMoitra** [SemanticApplicationDesignLanguage/sadl#789] [Q&A]
@glguy brought this to my attention.

I am using a user-defined data type in SADL as follows
      Range is a type of float [0,1].
The OWL that is generated is as follows

  <rdfs:Datatype rdf:ID="Range">
    <owl:equivalentClass>
      <rdfs:Datatype>
        <owl:withRestrictions rdf:parseType="Collection">
          <rdf:Description>
            <xsd:maxInclusive rdf:datatype="http://www.w3.org/2001/XMLSchema#float"
            >1</xsd:maxInclusive>           
            <xsd:minInclusive rdf:datatype="http://www.w3.org/2001/XMLSchema#float"
            >0</xsd:minInclusive>
          </rdf:Description>
        </owl:withRestrictions>
        <owl:onDatatype rdf:resource="http://www.w3.org/2001/XMLSchema#float"/>
      </rdfs:Datatype>
    </owl:equivalentClass>
  </rdfs:Datatype>

However the following is what works in Protege; notice the extra "</rdf:Description> <rdf:Description>"

  <rdfs:Datatype rdf:ID="Range">
    <owl:equivalentClass>
      <rdfs:Datatype>
        <owl:withRestrictions rdf:parseType="Collection">
          <rdf:Description>
            <xsd:maxInclusive rdf:datatype="http://www.w3.org/2001/XMLSchema#float"
            >1</xsd:maxInclusive>
          </rdf:Description>
          <rdf:Description>
            <xsd:minInclusive rdf:datatype="http://www.w3.org/2001/XMLSchema#float"
            >0</xsd:minInclusive>
          </rdf:Description>
        </owl:withRestrictions>
        <owl:onDatatype rdf:resource="http://www.w3.org/2001/XMLSchema#float"/>
      </rdfs:Datatype>
    </owl:equivalentClass>
  </rdfs:Datatype>

Should SADL be generating OWL differently or should Protege be ok with what SADL is generating?

**Comment by crapo** [comment]
After looking into it a bit I conclude that Protege is probably correct. The range of owl:withRestrictions is an rdf:List, and Protege's construct conforms with that. Plus I found a least one example from another source that is similar. I'll look into having SADL create the same.
